### PR TITLE
The error message is not user-friendly when adding duplicate permissi…

### PR DESCRIPTION
**Pull request by zhouchunhai**
…ons. (#12773)

## What is the purpose of the change

The error message is not user-friendly when adding duplicate permissions.

### CODE REVIEW

**Comment by wuyfee** [comment]
$\color{green}{SUCCESS}$
[DETAILS](https://github.com/alibaba/nacos/actions/runs/11609496226)
 ✅ - docker:   success 
 ✅ - deploy (standalone & cluster & standalone_auth):   success 
 ✅ - e2e-java-test (standalone & cluster & standalone_auth):   success 
 ✅ - e2e-go-test (standalone & cluster):   success 
 ✅ - e2e-cpp-test (standalone & cluster):   success 
 ✅ - e2e-csharp-test (standalone & cluster):   success 
 ✅ - e2e-nodejs-test (standalone & cluster):   success 
 ✅ - e2e-python-test (standalone & cluster):   success 
 ✅ - clean (standalone & cluster & standalone_auth):   success

**Comment by KomachiSion** [comment]
Is relatived with #12803?

**Comment by zhouchunhai** [comment]
It's relatived with [ #12773 ](https://github.com/alibaba/nacos/issues/12773)

**Comment by KomachiSion** [comment]
But I think add these two http api can't solve this problem.

the console ui will not call these two api.

**Comment by zhouchunhai** [comment]
My colleague has already submitted a Pull Request (PR). The console ui will call the new API to check if the permission already exists.

**Comment by KomachiSion** [comment]
> My colleague has already submitted a Pull Request (PR). The console ui will call the new API to check if the permission already exists.

So I ask you whether your pr is relative with https://github.com/alibaba/nacos/pull/12803 ?

**Comment by zhouchunhai** [comment]
yes, it's relative with #12803

File: plugin-default-impl/nacos-default-auth-plugin/src/main/java/com/alibaba/nacos/plugin/auth/impl/controller/PermissionController.java [hunk 1]
```diff
@@ -105,4 +105,18 @@ public Object deletePermission(@RequestParam String role, @RequestParam String r
...
+     * Judge whether a permission is duplicate.
+     *
+     * @param role     the role
+     * @param resource the related resource
+     * @param action   the related action
+     * @return true if duplicate, false otherwise
+     */
+    @GetMapping
+    @Secured(resource = AuthConstants.CONSOLE_RESOURCE_NAME_PREFIX + "permissions", action = ActionTypes.READ)
+    public Boolean isDuplicatePermission(@RequestParam String role, @RequestParam String resource, @RequestParam String action) {
```
Comment (KomachiSion) [inline plugin-default-impl/nacos-default-auth-plugin/src/main/java/com/alibaba/nacos/plugin/auth/impl/controller/PermissionController.java:119]: Should we response with Result<Boolean>?

**Review by KomachiSion — CHANGES_REQUESTED** [review]

File: plugin-default-impl/nacos-default-auth-plugin/src/main/java/com/alibaba/nacos/plugin/auth/impl/controller/PermissionController.java (hunk 1 above)
Comment (zhouchunhai) [reply plugin-default-impl/nacos-default-auth-plugin/src/main/java/com/alibaba/nacos/plugin/auth/impl/controller/PermissionController.java:119]: if not response with a result, how can ui judge whether a permission is duplicate

**Comment by wuyfee** [comment]
$\color{red}{FAILURE}$
[DETAILS](https://github.com/alibaba/nacos/actions/runs/11945535114)
 ✅ - docker:   success 
 ✅ - deploy (standalone & cluster & standalone_auth):   success 
 ✅ - e2e-java-test (standalone & cluster & standalone_auth):   success 
 ✅ - e2e-go-test (standalone & cluster):   success 
 ✅ - e2e-cpp-test (standalone & cluster):   success 
 ❌ - e2e-csharp-test (standalone & cluster):   failure 
 ✅ - e2e-nodejs-test (standalone & cluster):   success 
 ✅ - e2e-python-test (standalone & cluster):   success 
 ✅ - clean (standalone & cluster & standalone_auth):   success

File: plugin-default-impl/nacos-default-auth-plugin/src/main/java/com/alibaba/nacos/plugin/auth/impl/controller/PermissionController.java (hunk 1 above)
Comment (KomachiSion) [reply plugin-default-impl/nacos-default-auth-plugin/src/main/java/com/alibaba/nacos/plugin/auth/impl/controller/PermissionController.java:119]: I mean response with class Result not only return an Boolean.

**Review by KomachiSion — CHANGES_REQUESTED** [review]

**Comment by wuyfee** [comment]
$\color{green}{SUCCESS}$
[DETAILS](https://github.com/alibaba/nacos/actions/runs/12082134595)
 ✅ - docker:   success 
 ✅ - deploy (standalone & cluster & standalone_auth):   success 
 ✅ - e2e-java-test (standalone & cluster & standalone_auth):   success 
 ✅ - e2e-go-test (standalone & cluster):   success 
 ✅ - e2e-cpp-test (standalone & cluster):   success 
 ✅ - e2e-csharp-test (standalone & cluster):   success 
 ✅ - e2e-nodejs-test (standalone & cluster):   success 
 ✅ - e2e-python-test (standalone & cluster):   success 
 ✅ - clean (standalone & cluster & standalone_auth):   success

**Comment by KomachiSion** [comment]
Hello, you new code ci can't pass, please check the detail log and fix it.

**Comment by wuyfee** [comment]
$\color{green}{SUCCESS}$
[DETAILS](https://github.com/alibaba/nacos/actions/runs/12151890159)
 ✅ - docker:   success 
 ✅ - deploy (standalone & cluster & standalone_auth):   success 
 ✅ - e2e-java-test (standalone & cluster & standalone_auth):   success 
 ✅ - e2e-go-test (standalone & cluster):   success 
 ✅ - e2e-cpp-test (standalone & cluster):   success 
 ✅ - e2e-csharp-test (standalone & cluster):   success 
 ✅ - e2e-nodejs-test (standalone & cluster):   success 
 ✅ - e2e-python-test (standalone & cluster):   success 
 ✅ - clean (standalone & cluster & standalone_auth):   success

**Comment by DirtyBit64** [comment]
角色管理模块添加重复的绑定关系（角色-用户）也存在同样的问题

**Review by KomachiSion — APPROVED** [review]

**Comment by KomachiSion** [comment]
> 角色管理模块添加重复的绑定关系（角色-用户）也存在同样的问题

Welcome to contribute.

**Comment by DirtyBit64** [comment]
> > 角色管理模块添加重复的绑定关系（角色-用户）也存在同样的问题
> 
> Welcome to contribute.

好的，已提交PR [#12926](https://github.com/alibaba/nacos/pull/12926)

**Pull request merged**